wp/lib/bap_wp/src/precondition.mli

```diff
@@ -122,6 +122,9 @@ val caller_saved_regs : Bap.Std.Arch.t -> Bap.Std.Var.t list
 (** Obtains the callee-saved registers for a given architecture. *)
 val callee_saved_regs : Bap.Std.Arch.t -> Bap.Std.Var.t list
 
+(** FILL ME IN **)
+val user_func_spec : string -> string -> string -> Bap.Std.Sub.t -> Bap.Std.Arch.t -> Env.fun_spec option
```

Comment:
Since this function is exposed in the mli, I think it's a good idea to have labeled arguments for the strings, so we know what each string is supposed to be. Something like `name:string -> pre:string -> post:string` maybe?